src/models/db/notification.js

```diff
@@ -1,26 +1,89 @@
 'use strict'
 
-module.exports = NotificationFactory
-
-const Container = require('constitute').Container
-const Model = require('five-bells-shared').Model
-const PersistentKnexModelMixin = require('five-bells-shared').PersistentKnexModelMixin
-const CaseFactory = require('./case')
 const knex = require('../../lib/knex').knex
+const _ = require('lodash')
+const assert = require('assert')
+
+const TABLE_NAME = 'N_NOTIFICATIONS'
+
+const NOTIFICATION_STATUS_IDS = {
+  1: 'pending',
+  2: 'delivered',
+  3: 'failed'
+}
+
+const NOTIFICATION_DESCRIPTIONS = _.invert(NOTIFICATION_STATUS_IDS)
 
-NotificationFactory.constitute = [Container, CaseFactory]
-function NotificationFactory (container, Case) {
-  class Notification extends Model {
-    static convertFromPersistent (data) {
-      delete data.created_at
-      delete data.updated_at
-      return data
-    }
+function knexOrTransaction (knex, options) {
+  return !options ? knex : (!options.transaction ? knex : options.transaction)
+}
+
+function saveNotification (notificationObject, options) {
```

Comment:
I think we always know whether we are updating or inserting, so I think we can split this into two methods.